src/utilities/modules/vaccines.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Utility and helper methods for the vaccine
+ * module.
+ */
+
+/**
+ * Method used with Array.prototype.sort for sensorLogs
+ * Sorts from oldest to newest.
+ * @param  {SensorLog} sensorLogA
+ * @param  {SensorLog} sensorLogB
+ * @param  {Date}      sensorLogA.timestamp
+ * @param  {Date}      sensorLogB.timestamp
+ * @return {Number}    1 for A>B, -1 for A<B, 0 for A=B
+ */
+const sortSensorLogs = (
+  { timestamp: timestampA = new Date(-8640000000000000) },
+  { timestamp: timestampB = new Date(-8640000000000000) }
+) => {
+  if (timestampA > timestampB) return 1;
+  if (timestampA < timestampB) return -1;
+  return 0;
+};
+
+/**
+ * Extracts breaches from a set of sensor logs.
+ * Breach: Sequential sensorLog objects whose temperature is outside
+ * the lower or uppound limits, form a breach. Each array returned
+ * contains all sensorLogs which form a breach INCLUDING the sensorLog
+ * prior to the breach, and after the breach, if they exist. There may
+ * be duplicate non-breach sensorLogs as breach delimiters if there is
+ * only one sensorLog between breaches.
+ *
+ * Method sorts the sensorLogs by timestamp, iterates and pushes onto a
+ * stack. Only one unBreached sensorLog is kept at the tail of the stack.
+ * All breached sensorLogs are pushed, until a non breached log, where this
+ * marks a breach has been found. Store these sensorLogs and reset the stack.
+ *
+ * @param  {Realm.results/array} sensorLogs array of sensorLog objects
+ * @return {Array<Realm.results<SensorLog>} A 2D array of Realm.results. Example:
+ * [ [ sensorLog1, sensorLog2 .. ], [ sensorLog1, sensorLog2, sensorLog3, .. ]]
+ * where each element is a Realm.results<SensorLog>
+ */
+export const extractBreaches = ({ sensorLogs = [], database }) => {
+  if (!(database && sensorLogs.length > 0)) return [];
+
+  const sensorLogStack = [];
+  const breaches = [];
+
+  // Sort the sensor logs by timestamp.
+  sensorLogs.sort(sortSensorLogs);
```

Comment:
I don't think `.sort` is defined for Realm collections?

Reply:
69db3a26b57771c4457fff893da0579fdf5901b6 

(I *think* it works though, I can't remember if I actually tested this explicitly)